spark-3.5.yaml

```diff
@@ -20,370 +33,240 @@ environment:
-            - python-3.11
-            - pyspark-scala-2.13
+            - ${{package.name}}-scala-2.13-compat
+            - R
+            - openjdk-${{vars.test-java-version}}-default-jvm
+            - python3
+            - wait-for-it
         environment:
-          LANG: en_US.UTF-8
-          SCALA_VERSION: 2.13
-          JAVA_HOME: /usr/lib/jvm/${{range.value}}
-          SPARK_LOCAL_IP: 127.0.0.1
-          SPARK_LOCAL_HOSTNAME: localhost
-          SPARK_HOME: /usr/lib/spark
-          SPARK_USER: test
+          SPARK_CONF: "--conf spark.ssl.enabled=true --conf spark.ssl.protocol=TLSv1.3 --conf spark.network.crypto.enabled=true --conf spark.authenticate=true --conf spark.authenticate.secret=AbCdEfG0%12345678NotReal --conf spark.io.encryption.enabled=true"
+          SPARK_HOME: "/usr/lib/spark"
+          JAVA_HOME: "/usr/lib/jvm/default-jvm"
       pipeline:
-        - name: Test ${{package.name}} with different versions of OpenJDK
-          pipeline:
-            - name: Test if the Scala versions are correct
-              runs: ls /usr/lib/spark/jars/scala-* | grep -q $SCALA_VERSION
-            - name: Check spark-shell --version
-              runs: /usr/lib/spark/bin/spark-shell --version
-            - name: Check spark-submit --version
-              runs: /usr/lib/spark/bin/spark-submit --version
-            - name: Check pyspark --version
-              runs: /usr/lib/spark/bin/pyspark --version
-            - name: Check spark-sql --version
-              runs: /usr/lib/spark/bin/spark-sql --version
-            - name: Test entrypoint
-              runs: timeout 10 /usr/lib/spark/bin/spark-shell > spark_log.txt 2>&1 || [ $? -eq 143 ] && grep "Spark session available" spark_log.txt && exit_code=0 || exit_code=$? && echo $exit_code
-            - name: Run a simple Scala test script
-              runs: |
-                cat <<EOF > SimpleJob.scala
-                val data = Seq(1, 2, 3, 4, 5)
-                val rdd = spark.sparkContext.parallelize(data)
-                val sum = rdd.reduce(_ + _)
-                assert(sum == 15)
-                EOF
-                cat SimpleJob.scala | /usr/lib/spark/bin/spark-shell --conf spark.jars.ivy=/tmp/.ivy --master local[*]
-            - name: Perform SQL query on DataFrame in Scala
-              runs: |
-                cat <<EOF > SQLTest.scala
-                import org.apache.spark.sql.SparkSession
-                val spark = SparkSession.builder.appName("SQLTest").getOrCreate()
-                import spark.implicits._
-                val df = Seq((1, "Alice"), (2, "Bob")).toDF("id", "name")
-                df.createOrReplaceTempView("people")
-                val result = spark.sql("SELECT name FROM people WHERE id = 2")
-                assert(result.count() == 1 && result.first().getString(0) == "Bob")
-                EOF
-                cat SQLTest.scala | /usr/lib/spark/bin/spark-shell --conf spark.jars.ivy=/tmp/.ivy --master local[*]
```

Comment:
I don't see a scala test anymore, is that intentional? The SQL, R, and Python tests look good

Reply:
Intentional, the Spark shell itself is already leveraging Scala and we can only have one running at a given time. Unless we want to track the PID, kill the running shell, and re-instantiate it, but I don't believe that's as good of a test as the shell already actively running every app we submit